libr/syscall/d/meson.build

```diff
@@ -21,29 +21,43 @@ sdb_files = [
   'windows-x86-64'
 ]
 
-sdbtmp_gen_path = '@0@/@1@'.format(meson.current_source_dir(), 'gen.sh')
-sdbtmp_gen_cmd = [
+sdb_pre_script = '''#script
+import re
+import sys
+inf = open(sys.argv[1])
+outf = open(sys.argv[2], 'w')
+for line in inf:
+    if not line.startswith('_') and '=' in line:
+        arr = re.split('=|,', line)
+        print('%s.%s=%s' % (arr[1], arr[2], arr[0]), file=outf)
+    outf.write(line)
+inf.close()
+outf.close()
+'''
+
+sdb_pre_cmd = [
   py3_exe,
   '-c',
-  'from sys import argv; __import__("os").system("@0@ <%s >%s" % (argv[1], argv[2]))'.format(sdbtmp_gen_path),
+  sdb_pre_script,
   '@INPUT@',
   '@OUTPUT@'
 ]
 
 foreach file : sdb_files
+  infile = run_command(sdb_symlink_cmd + [join_paths(meson.current_source_dir(),'@0@.sdb.txt'.format(file))]).stdout().strip()
```

Comment:
why is this necessary? i'm not sure i understand what it does.

Reply:
To resolve "symlinks" on Windows. By default git convert symlinks to a text file with a path to the file it points to.